fendermint/app/src/app.rs

```diff
@@ -194,6 +196,7 @@ where
         + Codec<FvmStateParams>,
     DB: KVWritable<S> + KVReadable<S> + 'static + Clone,
     SS: Blockstore + 'static + Clone,
+    P: ParentFinalityProvider,
```

Comment:
By the way just wondering if we really need to make this generic if we are using STM, are there going to be any other implementations? Like the `ResolverPool` is passed directly, but in itself it's just a communication area, so any change of behaviour would be to whoever is feeding it on the other side.